### PR TITLE
SCICD-605: Threading Problems

### PR DESCRIPTION
**Pull request by larry-tordsen**
This fixes some threading problems.

## Summary and Scope
Simplified the threading some, and made it more efficient.

## Issues and Related PRs

* SCICD-605 https://jira-pro.its.hpecorp.net:8443/browse/SCICD-605

### Tested on:

  * gamora

### CODE REVIEW

File: lib/Activity.py [hunk 1]
```diff
@@ -318,7 +318,17 @@ def get_next_workflow(self, sessionid):
                 self.config.logger.error(f"Unable to get session {sessionid}: {e}")
                 sys.exit(1)
 
-            session = rsession.json()
+            # FIXME:
```
Comment (larry-tordsen) [inline lib/Activity.py:321]: I'm not sure about this -- it might be worth it's own ticket.

File: lib/Activity.py [hunk 2]
```diff
@@ -768,7 +775,7 @@ def resume(self):
             self.config.logger.error(f"Unable to get activity {self.name} from backend.")
             return
 
-        if last_activity["state"] == 'in_progress':
+        if last_activity["state"] == 'in_progress' and last_activity["status"].lower() not in ["succeeded", "failed"]:
```
Comment (larry-tordsen) [inline lib/Activity.py:778]: This might need future tweaking.  Not sure about which `status`es indicate a status of done.

**Comment by deantroe** [comment]
@larry-tordsen if you don't mind me abusing your PR, could you fix this parser line (since upper-case is no longer valid):
```
        Activity names must only contain letters (A-Za-z), numbers (0-9), periods (.), and dashes (-).
```

**Comment by larry-tordsen** [comment]
> Activity names must only contain letters

Done.  It now reads:
```
Activity names must contain only lowercase letters (a-z), numbers (0-9), periods (.), and dashes (-).
Can also be set via the IUF_ACTIVITY environment variable
```

**Review by joshua-plautz — APPROVED** [review]

**Pull request merged**